diff --git a/erts/etc/unix/erl.src.src b/erts/etc/unix/erl.src.src
index 959c099e8f..0983132ac1 100644
--- a/erts/etc/unix/erl.src.src
+++ b/erts/etc/unix/erl.src.src
@@ -1,9 +1,9 @@
 #!/bin/sh
 #
 # %CopyrightBegin%
-# 
-# Copyright Ericsson AB 1996-2016. All Rights Reserved.
-# 
+#
+# Copyright Ericsson AB 1996-2020. All Rights Reserved.
+#
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
 # You may obtain a copy of the License at
@@ -15,10 +15,15 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-# 
+#
 # %CopyrightEnd%
 #
-ROOTDIR="%FINAL_ROOTDIR%"
+if [ -z "$ERL_ROOTDIR" ]
+then
+    ROOTDIR="%FINAL_ROOTDIR%"
+else
+    ROOTDIR="$ERL_ROOTDIR"
+fi
 BINDIR=$ROOTDIR/erts-%VSN%/bin
 EMU=%EMULATOR%%EMULATOR_NUMBER%
 PROGNAME=`echo $0 | sed 's/.*\///'`
diff --git a/erts/etc/unix/start.src b/erts/etc/unix/start.src
index bdd146951f..4cf0e899e4 100644
--- a/erts/etc/unix/start.src
+++ b/erts/etc/unix/start.src
@@ -1,9 +1,9 @@
 #!/bin/sh
 #
 # %CopyrightBegin%
-# 
-# Copyright Ericsson AB 1996-2016. All Rights Reserved.
-# 
+#
+# Copyright Ericsson AB 1996-2020. All Rights Reserved.
+#
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
 # You may obtain a copy of the License at
@@ -15,7 +15,7 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-# 
+#
 # %CopyrightEnd%
 #
 # This program invokes the erlang emulator by calling run_erl.
@@ -25,7 +25,12 @@
 #
 # Usage: start [Data]
 #
-ROOTDIR=%FINAL_ROOTDIR%
+if [ -z "$ERL_ROOTDIR" ]
+then
+    ROOTDIR="%FINAL_ROOTDIR%"
+else
+    ROOTDIR="$ERL_ROOTDIR"
+fi
 
 if [ -z "$RELDIR" ]
 then
